cosmwasm/README.md

```diff
@@ -1,45 +1,92 @@
-# Deploy
+# Intro
 
-First build the contracts
+This folder contains pyth contract on CosmWasm and utilities to deploy it in CosmWasm chains.
 
+# Deployment
+
+Deploying a contract in terra consists of two steps:
+1. Uploading the code. This step will give you a code id.
+2. Optionally create a new contract or migrate an existing one:
+    1. Creating a new contract which has an address with a code id as its program.
+    2. Migrating an existing contract code id to the new code id.
+
+This script can do both steps at the same time. Read below for the details.
+
+## Uploading the code
+
+First build the contracts:
 
 ``` sh
-docker build -f Dockerfile.build -o artifacts .
+bash build.sh
 ```
 
-Then, for example, to deploy `token_bridge.wasm`, run in the `tools` directory
+This command will builds and saves all the contracts in the `artifact` directory.
+
+Then, for example, to deploy `pyth_bridge.wasm`, run in the `tools` directory:
 
 ``` sh
-npm ci
-node deploy_single.js --network mainnet --artifact ../artifacts/token_bridge.wasm --mnemonic "..."
+npm ci # Do it only once to install required packages
+npm run deploy-pyth -- --network testnet --artifact ../artifacts/pyth_bridge.wasm --mnemonic "..."
 ```
 
-which will print something along the lines of
+which will print something along the lines of:
 
 ``` sh
-Storing WASM: ../artifacts/token_bridge.wasm (367689 bytes)
+Storing WASM: ../artifacts/pyth_bridge.wasm (367689 bytes)
 Deploy fee:  88446uluna
 Code ID:  2435
 ```
 
-# Migrate
+If you do not pass any additional arguments to the script it will only upload the code and returns the code id. If you want to create a 
+new contract or upgrade an existing contract you should pass more arguments that are described below.
 
-## Mainnet
+## Instantiating new contract
+If you want instantiate a new contract after your deployment pass `--instantiate` argument to the above command.
+It will upload the code and with the resulting code id instantiates a new pyth contract:
```

Comment:
general comment: I think you overuse the word "it". Replacing "it" with a more specific referring expression (e.g., "this command") will improve your writing clarity, as it is easier for the reader to identify what you are referring to.